samtranslator/model/api/http_api_generator.py

```diff
@@ -468,7 +479,8 @@ def _set_default_authorizer(self, open_api_editor, authorizers, default_authoriz
         if not default_authorizer:
             return
 
-        if not authorizers.get(default_authorizer):
+        # The AWS_IAM authorizer is built-in and does not need to be defined in the template as an authorizer.
```

Comment:
We require this for all other Auths. Why should AWS_IAM be any different?

Reply:
Because AWS_IAM is a "built in" auth type. It should not be customized and there only ever needs to be a single one of them. That said I did take the approach you are suggesting in my alternate implementation: https://github.com/aws/serverless-application-model/pull/1876

Reply:
And we already have a precedent for another built-in authorizer: `NONE`. The `NONE` authorizer could have been implemented as some kind of authorizer you had to define but which did nothing, which would make it like all the other authorizers, but instead it was defined as a special built-in authorizer. I implemented something similar with the AWS_IAM authorizer.